tests/grids/test_refinement.py

```diff
@@ -38,25 +40,88 @@ def test_grid_perturbation_1d_internal_nodes_fixed(self):
+                grid.
+
+        """
+
+        # The two grids should cover the same domain.
+        assert np.sum(g.cell_volumes) == np.sum(g_ref.cell_volumes)
+
+        # If the ratio is given, all cells should be refined by this ratio.
+        if ratio is not None:
+            assert np.allclose(g_ref.num_cells / g.num_cells, ratio)
+
+        # There should be as many faces as nodes in a 1d grid. Check that the face-node
+        # map is 1-1, and that there are as many node coordinates as there are nodes.
+        assert g_ref.face_nodes.shape[0] == g_ref.face_nodes.shape[1]
+        assert g_ref.face_nodes.shape[0] == g_ref.nodes.shape[1]
+
+        # These are 1d grids, thus the number of boundary faces should be the same in
+        # the two grids. Exploit this to check the refined cell-face mapping, which has
+        # a somewhat non-trivial construction.
+        num_boundary_g = np.sum(np.sum(g.cell_faces, axis=1) != 0)
+        num_boundary_g_ref = np.sum(np.sum(g_ref.cell_faces, axis=1) != 0)
+        assert num_boundary_g == num_boundary_g_ref
+
+    @pytest.mark.parametrize("ratio", [2, 3])
+    def test_refinement_grid_1D(self, ratio: int):
+        """Test the refinement of a 1D grid
+
+        Parameters:
+            ratio: The ratio of the number of cells in the refined grid to the original
+                grid.
+
+        """
+        # First create four grids with increasing complexity, then refine each of them
+        # and verify that the refined grid has the expected properties.
+
+        # A simple uniform grid.
+        g_0 = pp.TensorGrid(np.array([0, 1, 2, 3, 4]))
+
+        # A grid with non-uniform spacing.
+        g_1 = pp.TensorGrid(np.array([0, 1, 2, 2.5, 3, 4]))
+
+        # A grid with a permuted node/face (they can be considered identical in 1d) and
+        # cell ordering: The nodes lie along the x-axis, but in the order [1, 0, 2, 3].
+        # The cells are ordered (with increasing x-coordinates) [1, 0, 2].
+        x = np.array([[1, 0, 2, 3], [0, 0, 0, 0], [0, 0, 0, 0]])
+        fn = sps.identity(4, format="csr")
+        cf = sps.csc_array(np.array([[-1, 1, 0], [0, -1, 0], [1, 0, -1], [0, 0, -1]]))
+        g_2 = pp.Grid(dim=1, nodes=x, face_nodes=fn, cell_faces=cf, name="")
+
+        # A grid where one node (coordinate x=2) has been split, as would happen if two
+        # 1d grids representing fractures intersect.
+        x = np.array([[0, 1, 2, 2, 3], [0, 0, 0, 0, 0], [0, 0, 0, 0, 0]])
+        fn = sps.identity(5, format="csr")
+        cf = sps.csc_array(
+            np.array([[-1, 1, 0, 0, 0], [0, -1, 1, 0, 0], [0, 0, 0, -1, 1]]).T
         )
+        g_3 = pp.Grid(dim=1, nodes=x, face_nodes=fn, cell_faces=cf, name="")
+
+        for g in [g_0, g_1, g_2, g_3]:
+            g.compute_geometry()
+            g_ref = pp.refinement.refine_grid_1d(g, ratio)
+            self.compare_refined_grids(g, g_ref, ratio)
```

Comment:
Is this check really convincing that the refinement does what it should? I would consider adding one "bad" test for g_0, and see whether g_ref coincides with [0, 0.5, 1, 1.5, ...] as well as [0, 0.33, 0.67, 1, ...] - a bit more like the old tests, where at least nodes were compared. At the moment merely shapes are tested but not really topology.

Reply:
Added a test that the node coordinates are as expected. 

I also modified the documentation somewhat to make the rational behind the testing clearer: It is not meaningful to do a detailed check of the topology of the refined grid, as no guarantees are given for this beyond its sanity. In particular the ordering of nodes and cells can differ significantly between the old and refined grid.

Reply:
Thanks for the clarification. I think the newly added test does nevertheless not hurt. Thanks for adding it.